lib/hoplon/crypto.ex

```diff
@@ -0,0 +1,195 @@
+defmodule Hoplon.Crypto do
+  @doc """
+  All the wrapper tools for low-level cryptography, specific to Hoplon's use-case.
+  """
+
+  # see https://crypto.stackexchange.com/a/10825 for why this is here
+  @standard_rsa_public_exponent 65537
+  @expected_key_bit_size 4096
+
+  # if the message isn't digested, the signature fails for bigger messages (but not all that huge)
+  @digest_type :sha512
+  @rsa_pk_sign_verify_opts []
+
+  alias Hoplon.Error
+  require Hoplon.Crypto.Records
+  alias Hoplon.Crypto.Records
+
+  defguard is_public_key(maybe_rsa_public_key)
+           when is_tuple(maybe_rsa_public_key) and :RSAPublicKey == elem(maybe_rsa_public_key, 0)
+
+  defguard is_private_key(maybe_rsa_private_key)
+           when is_tuple(maybe_rsa_private_key) and
+                  :RSAPrivateKey == elem(maybe_rsa_private_key, 0)
+
+  @spec generate_private_key() :: Records.rsa_private_key()
+  def generate_private_key() do
+    :public_key.generate_key({:rsa, @expected_key_bit_size, @standard_rsa_public_exponent})
+  end
+
+  @spec build_public_key(Records.rsa_private_key()) :: Records.rsa_public_key()
+  def build_public_key(private_key) do
```

Comment:
Add `validate_public_key` and `validate_private_key` functions that return error tuples and verify the keys fill our requirements (RSA, 4096 bit length, anything else?)